deploy/olm-catalog/service-assurance-operator/0.1.0/infra.watch_serviceassurances_crd.yaml

```diff
@@ -42,6 +42,12 @@ spec:
             eventsEnabled:
               description: Whether the Service Assurance Operator should enable components related to events collection and storage.
               type: boolean
+            haEnabled:
+              description: Whether to deploy the services in HA mode.
+              type: boolean
+            esNodeCount:
+              description: Number of elastic search instances to deploy in HA mode (ignored unless haEnabled == True)
+              type: string
```

Comment:
integer?

Reply:
This was my thought also, however, I think it has to correspond with deploy/olm-catalog/service-assurance-operator/0.1.0/service-assurance-operator.v0.1.0.clusterserviceversion.yaml, and I'm not sure where those types come from, when I find that out, this should be updated to and integer field

Reply:
As noted earlier, this is a compiled file from the contents of `deploy/`. You would compile a new CSV with `operator-sdk olm-catalog gen-csv --update-crds --csv-version 0.1.0`

Reply:
Should we gitignore the compiled files, then?

Reply:
No because they are only partially compiled :)

Reply:
So basically when you run the `gen-csv` it'll update the `alm-example` section with the values in the `cr` file within `deploy/crds/`. It'll also copy over the `_crd` file from the `deploy/crds/` directory into the `olm-catalog/service-assurance-operator/0.1.0` directory.

The rest of the CSV is basically then maintained by hand. If this were a Go Operator, then more things could be handled in the code and the `gen-csv` command would have more hints to generating more of the CSV file itself.

When you're doing testing, the `deploy/crds` directory is used, like when we run the `operator-sdk scorecard`. It's expected that the `deploy/crds` directory exists with your CRD and CR files for handling a deployment. The `olm-catalog` directory is effectively a package manifest for the deployment of the Operator.

Some things are duplicated, but those that are only exist in a single location, and the duplicated data is generated by `operator-sdk`. Another example is if you change the `deploy/role.yaml` to modify the roles available for the Operator, that will be updated in the `CSV` file when you run `gen-csv` as well.

Reply:
Thanks!